### ISSUE TITLE
How to set environment variable in a task?

### ISSUE THREAD

**Issue by tajalarefin** [gogradle/gogradle#173]
My code uses some environment variables. I have written test cases which will invoke code to look for environment variables. I tried to add the test task as:
`test {
    environment 'SERVER_NAME': 'myserver'
    commandLine 'gradle', 'goTest'
}`

But it says method environment() not found. Is there a way I can specify the environment variable in test task?

**Comment by blindpirate** [comment]
Unfortunately we can't set environment only for test task. However, there's a workaround:

```
task setEnvVar {
     doLast {
            System.setenv('SERVER_NAME','myserver')
     }
}

goTest.dependsOn setEnvVar
```

I'll try to add this support in next release.

**Comment by tajalarefin** [comment]
I tried to use the way mentioned but getting following error:
`No signature of method: static java.lang.System.setenv() is applicable for argument types: (java.lang.String, java.lang.String) values: [SERVER_NAME, myserver]
  Possible solutions: getenv(), getenv(java.lang.String), setIn(java.io.InputStream), every(), setErr(java.io.PrintStream), setOut(java.io.PrintStream)
`

**Comment by blindpirate** [comment]
Sorry for misleading you, I was wrong, this doesnt work. I'm out for
vacation so it seems no way to work around. Will solve in next release,
sorry for that.

**Comment by anuraaga** [comment]
Thanks @blindpirate looking forward to being able to configure the environment for tests :)

In the meantime, @tajalarefin I have come up with this ridiculous hack based on reflection in case you're interested in it
```groovy
class EnvironmentSettingBuildManager implements com.github.blindpirate.gogradle.build.BuildManager {

    com.github.blindpirate.gogradle.build.BuildManager delegate

    EnvironmentSettingBuildManager(com.github.blindpirate.gogradle.build.BuildManager delegate) {
        this.delegate = delegate
    }

    @Override
    void prepareProjectGopathIfNecessary() {
        delegate.prepareProjectGopathIfNecessary()
    }

    @Override
    String getGopath() {
        return delegate.getGopath()
    }

    @Override
    List<Path> getGopaths() {
        return delegate.getGopaths()
    }

    @Override
    int go(List<String> args, Map<String, String> env) {
        return delegate.go(args, env)
    }

    @Override
    int go(List<String> args, Map<String, String> env, Consumer<String> stdoutLineConsumer, Consumer<String> stderrLineConsumer) {
        return delegate.go(args, env, stderrLineConsumer, stderrLineConsumer)
    }

    @Override
    int go(List<String> args, Map<String, String> env, Consumer<String> stdoutLineConsumer, Consumer<String> stderrLineConsumer, boolean continueWhenFail) {
        def overridden = [CGO_ENABLED: '0']
        return delegate.go(args, overridden, stdoutLineConsumer, stderrLineConsumer, continueWhenFail)
    }

    @Override
    int run(List<String> args, Map<String, String> env, Consumer<String> stdoutLineConsumer, Consumer<String> stderrLineConsumer) {
        return delegate.run(args, env, stdoutLineConsumer, stderrLineConsumer)
    }

    @Override
    int run(List<String> args, Map<String, String> env, Consumer<String> stdoutLineConsumer, Consumer<String> stderrLineConsumer, boolean continueWhenFail) {
        return delegate.run(args, env, stdoutLineConsumer, stderrLineConsumer, continueWhenFail)
    }
}

def buildManagerField = com.github.blindpirate.gogradle.task.go.GoTest.declaredFields[4]
buildManagerField.accessible = true
def defaultBuildManager = buildManagerField.get(tasks.goTest)
buildManagerField.set(tasks.goTest, new EnvironmentSettingBuildManager(defaultBuildManager))
```

**Comment by tajalarefin** [comment]
@blindpirate - Any update on setting Environment variable in goTest?

**Comment by RobTrifiletti** [comment]
I would also be very interested in setting environment variables in goTest

**Comment by tajalarefin** [comment]
Found an alternate way to test with env variable.

`task myTest (type: Exec) {
def cmd = "gradle goTest"
environment "SERVER_NAME", "my-server"
commandLine = cmd.tokenize()
}`

This will execute goTest with env variable.

**Comment by blindpirate** [comment]
Closed by `0.9`. Thanks for reporting.
